openpgp/internal/ecc/curve_info.go

```diff
@@ -9,75 +9,84 @@ import (
 	"github.com/ProtonMail/go-crypto/openpgp/internal/encoding"
 )
 
-type SignatureAlgorithm uint8
-
 type CurveInfo struct {
-	Name string
+	Type string
```

Comment:
The term "Type" here is a bit confusing to me, as I would expect it to be "ECDSA", or maybe "Weierstrass", but not "P256", for example. How about `GenName`, perhaps? (For "generic name" or "generation name" :)).

Reply:
I opted for `genName`, since it's used only internally we can make it private

Reply:
Sure, but I mean, the entire package is already under `internal`, so I don't think it matters too much; so I would just make it public like the others. I actually think it might be useful someday to have this generic name for display (as probably we want to show to the user "Curve448" instead of "x448/ed448")

Reply:
Public it is